packages/lodestar/src/api/impl/node/node.ts

```diff
@@ -43,30 +43,20 @@ export class NodeApi implements INodeApi {
     return this.sync.isSynced() ? "ready" : "syncing";
   }
 
-  public async getPeer(peerId: string): Promise<NodePeer | null> {
-    return (await this.getPeers()).find((peer) => peer.peerId === peerId) || null;
+  public async getPeer(peerIdStr: string): Promise<NodePeer | null> {
+    const connections = this.network.getConnectionsByPeer().get(peerIdStr);
+    if (!connections) return null; // Node has not seen this peer
+    return formatNodePeer(peerIdStr, connections);
   }
 
-  public async getPeers(state: string[] = [], direction: string[] = []): Promise<NodePeer[]> {
-    const nodePeers: NodePeer[] = [];
-    // if direction includes "inbound" or "outbound", it means we want connected peers
-    let peers =
-      (state.length === 1 && state[0] === "connected") || direction.length > 0
-        ? this.network.getPeers()
-        : this.network.getAllPeers();
-    peers = filterByStateAndDirection(peers, this.network, state, direction);
-    for (const peer of peers) {
-      const conn = this.network.getPeerConnection(peer.id);
-      nodePeers.push({
-        peerId: peer.id.toB58String(),
-        //TODO: figure out how to get enr of peer
-        enr: "",
-        lastSeenP2pAddress: conn ? conn.remoteAddr.toString() : "",
-        direction: conn ? conn.stat.direction : null,
-        state: conn ? getPeerState(conn.stat.status) : "disconnected",
-      });
-    }
-    return nodePeers;
+  public async getPeers(state?: PeerState[], direction?: PeerDirection[]): Promise<NodePeer[]> {
+    return Array.from(this.network.getConnectionsByPeer().entries())
+      .map(([peerIdStr, connections]) => formatNodePeer(peerIdStr, connections))
```

Comment:
It probably makes more sense to filter before map?

Reply:
The map run formats the objects from libp2p into our own API terms `state`, `direction`. So it's necessary to do that conversion first on all items to then be able to filter.